CMakeLists.txt

```diff
@@ -1,91 +1,60 @@
 #
 # uvw
 #
+cmake_minimum_required(VERSION 3.2)
 
 #
 # Building in-tree is not allowed (we take care of your craziness).
 #
 
-if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
     message(FATAL_ERROR "Prevented in-tree built. Please create a build directory outside of the source code and call cmake from there. Thank you.")
 endif()
 
 #
 # Project configuration
 #
 
-project(uvw)
-cmake_minimum_required(VERSION 3.4)
+project(uvw
+        VERSION 1.1.1)
 
 if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Debug)
-endif(NOT CMAKE_BUILD_TYPE)
+endif()
 
-set(PROJECT_NAME "uvw")
-set(PROJECT_VERSION_MAJOR 1)
-set(PROJECT_VERSION_MINOR 1)
-set(PROJECT_VERSION_PATCH 1)
-set(PROJECT_VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH})
 set(SETTINGS_ORGANIZATION "Michele Caini")
 set(SETTINGS_APPLICATION ${PROJECT_NAME})
 set(PROJECT_AUTHOR "Michele Caini")
-set(PROJECT_YEAR_COPYRIGHT "2017")
 set(PROJECT_AUTHOR_EMAIL "michele.caini@gmail.com")
 
-set(PROJECT_BUILD_MESSAGE ${PROJECT_NAME} " v" ${PROJECT_VERSION} " (" ${CMAKE_BUILD_TYPE} ")")
-set(COPYRIGHT_BUILD_MESSAGE "Copyright (c) " ${PROJECT_YEAR_COPYRIGHT} " " ${PROJECT_AUTHOR}  " <" ${PROJECT_AUTHOR_EMAIL} ">")
-
 message("*")
-message("* " ${PROJECT_BUILD_MESSAGE})
-message("* " ${COPYRIGHT_BUILD_MESSAGE})
+message("* ${PROJECT_NAME} v${PROJECT_VERSION} (${CMAKE_BUILD_TYPE})")
+message("* Copyright (c) 2017 ${PROJECT_AUTHOR} <${PROJECT_AUTHOR_EMAIL}>")
 message("*")
 
-if(APPLE)
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-undefined,dynamic_lookup")
-else(APPLE)
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
-endif(APPLE)
-
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 if(NOT WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
-endif(NOT WIN32)
+endif()
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -g -DDEBUG")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Os -DRELEASE")
 
-add_definitions(
-    -DPROJECT_NAME=${PROJECT_NAME}
-    -DPROJECT_VERSION=${PROJECT_VERSION}
-)
-
 #
 # CMake configuration
 #
 
-set(PROJECT_CMAKE_MODULES cmake/modules)
-set(PROJECT_CMAKE_IN cmake/in)
-set(PROJECT_BUILD_DIR build)
-set(PROJECT_DEPS_DIR deps)
-set(PROJECT_DOCS_DIR docs)
-set(PROJECT_SRC_DIR src)
-set(PROJECT_TEST_DIR test)
+set(PROJECT_CMAKE_IN ${uvw_SOURCE_DIR}/cmake/in)
+set(PROJECT_DEPS_DIR ${uvw_SOURCE_DIR}/deps)
+set(PROJECT_SRC_DIR ${uvw_SOURCE_DIR}/src)
 
-set(PROJECT_RUNTIME_OUTPUT_DIRECTORY bin)
-
-set(
-    CMAKE_MODULE_PATH
-    ${CMAKE_MODULE_PATH}
-    ${CMAKE_SOURCE_DIR}/${PROJECT_CMAKE_MODULES}
-)
-
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_RUNTIME_OUTPUT_DIRECTORY})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY bin)
 
 #
 # Enable test support using ctest
 #
 
-enable_testing()
```

Comment:
I'll remove `include(CTest)`.
I'll introduce `BUILD_TESTING` as an option because I'll need that for disabling test builds from the caller.